### PR TITLE
RangedEnemyComponent

### PR DESCRIPTION
**Pull request by kattalainen**
Made a range component for enemies. Try it out in sandbox andresScene. Additions/changes in the ai component is in progress (another task) to make it work better in the game scene.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/RangeEnemyComponent.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		void SetAttackInterval(float interval);
+		float GetAttackInterval() const;
+
+		void Attack(float3 direction);
+
+	private:
+		int m_Damage;
+		float m_Velocity;
+		float m_Scale;
+		int m_NrOfProjectiles = 0;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/Components/RangeEnemyComponent.h]: Better if initialise in constructor

File: StortSpelProjekt/Game/src/Gamefiles/Components/RangeEnemyComponent.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		float m_AttackInterval;
+		double m_TimeAccumulator;
+		std::string m_ParentName;
+
+		bool audioPlay;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/Components/RangeEnemyComponent.h]: m_AudioPlay

File: StortSpelProjekt/Sandbox/src/main.cpp [hunk 3]
```diff
@@ -1441,9 +1451,50 @@ Scene* AndresTestScene(SceneManager* sm)
 
 
     /* ---------------------- Enemy -------------------------------- */
-    // Enemyfactory used in the wrong way. 
-    //EnemyFactory enH(scene);
-    //entity = enH.AddEnemy("enemy", enemyModel, 1000, float3{ 0, 10, 20 }, L"Bruh", F_COMP_FLAGS::OBB | F_COMP_FLAGS::CAPSULE_COLLISION, 0, 0.5, float3{ 0, 0, 0 }, "player", 25.0f, 7.0f, 0.5f, 10.0f);
+#pragma region enemyRangeTest
+    Entity* ent = scene->AddEntity("enemyRangeTest");
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Sandbox/src/main.cpp:1448]: I assume this functionality will later be added to the enemy factory?

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
Runs well. Fix comments

File: StortSpelProjekt/Sandbox/src/main.cpp (hunk 3 above)
Comment (kattalainen) [reply StortSpelProjekt/Sandbox/src/main.cpp:1448]: Correct.

**Review by LeoWikstrom — DISMISSED** [review]

**Review by Schtekt — CHANGES_REQUESTED** [review]
Looks good, all scenes in sandbox and game works.
Don't forget the merge with develop though :)

**Review by LeoWikstrom — APPROVED** [review]

**Review by Schtekt — APPROVED** [review]

**Pull request merged**